### PR TITLE
#6 Provide more complete guidance on naming

### PR DESCRIPTION
**Pull request by tim-lyon-arup**
Incorporate proposals in issue https://github.com/arup-group/programming-styleguide/issues/6

### CODE REVIEW

File: cppguide.html [hunk 1]
```diff
@@ -4090,26 +4100,38 @@ <h3 id="General_Naming_Rules">General Naming Rules</h3>
 <a href="#Type_Names">type names</a>, and non-type template
 parameters should follow the rules for <a href="#Variable_Names">
 variable names</a>.
+</p>
+  
+<h3 id="Repository_Names">Repository Names</h3>
+<p>Repository names should be all lowercase and can include
+dashes (<code>-</code>). Oasys libraries should include an
+<code>oasys-</code> prefix. e.g. <code>oasys-useful-library</code></p>
```
Comment (ramaseshankannan) [inline cppguide.html:4108]: @tim-lyon-arup What is the reason to mandate libraries starting with `oasys-`?

**Review by ramaseshankannan — COMMENTED** [review]
Looks good, I've left a question

File: cppguide.html [hunk 2]
```diff
@@ -4090,26 +4100,38 @@ <h3 id="General_Naming_Rules">General Naming Rules</h3>
...
-<p>Filenames should be all lowercase and can include
-underscores (<code>_</code>) or dashes (<code>-</code>).
-Follow the convention that your
+<h3 id="File_Names">File Names</h3>
+<p>Newly created filenames should be all lowercase and can include
+underscores (<code>_</code>) but no dashes or spaces. This includes source files, project files, solution files and build output files</p>
 
-project uses. If there is no consistent
-local pattern to follow, prefer "_".</p>
+<p>Existing files should be only renamed as part of a planned refactoring task, not mixed with other changes.</p>
```
Comment (ftschindler) [inline cppguide.html:4118]: good one! :)

**Review by ftschindler — APPROVED** [review]

File: cppguide.html (hunk 1 above)
Comment (tim-lyon-arup) [reply cppguide.html:4108]: @ramaseshankannan This is for repo names not libraries, so we can find them more easily amongst the 3.6k+ repos in the arup-group org

File: cppguide.html [hunk 3]
```diff
@@ -4090,26 +4100,38 @@ <h3 id="General_Naming_Rules">General Naming Rules</h3>
 <a href="#Type_Names">type names</a>, and non-type template
 parameters should follow the rules for <a href="#Variable_Names">
 variable names</a>.
+</p>
+  
+<h3 id="Repository_Names">Repository Names</h3>
+<p>Repository names should be all lowercase and can include
+dashes (<code>-</code>). Oasys libraries should include an
```
Comment (tim-lyon-arup) [inline cppguide.html:4107]:
```suggestion
dashes (<code>-</code>). Oasys repos should include an
```

File: cppguide.html (hunk 1 above)
Comment (tim-lyon-arup) [reply cppguide.html:4108]: Updates to help clarify

**Review by ftschindler — DISMISSED** [review]

**Review by ramaseshankannan — APPROVED** [review]

**Review by ftschindler — APPROVED** [review]

**Pull request merged**